src/bundles/experiments.js

```diff
@@ -0,0 +1,97 @@
+const ACTIONS = {
+  EXP_TOGGLE: 'EXPERIMENTS_EXP_TOGGLE'
+}
+
+// setup experiments
+const EXPERIMENTS = {
+  npm: {
+    action: async () => null,
```

Comment:
`async () => doDesktopSettingsToggle('npmOnIpfs')` should do the trick

Reply:
Is that method available as a global in the electron app?

Reply:
Actions change state. We shouldn't need to await for an action, and shouldn't expect a useful value back from dispatching one.  We need to think about how we visualise the "we are enabling that action" state. Desktop will need to dispatch a new action once the experiment is enabled.

Reply:
No, `doDesktopSettingsToggle` comes from `ipfs-desktop` bundle and should only be called if `selectIsIpfsDesktop() === true`.

Reply:
@olizilla it already does. `selectDesktopSettings().npmOnIpfs`. It will always be up to date. Just like any other IPFS Desktop settings.

Reply:
@cwaring perhaps we could send the `store` object here too: https://github.com/ipfs-shipyard/ipfs-webui/pull/1048/files/e819b3ada40314ea3f25850fe1397b45bdf0b6a7#diff-74d797d2ce7950ecff72a587d6cf3f9dR86

So the action has access to all bundle actions

Reply:
Yes that is why I was asking because at the moment the actions defined here are simply callback functions that wait for a result before changing the state. It might be a better pattern to hook into the store and despatch something instead. Leaning on your preferred patterns for this so all hints welcome.

Reply:
@cwaring we can then perhaps expand the idea Oli had. @olizilla said we could pass an array of `ids` from `ipfsDesktop` with the experiments to enable. Perhaps it could be an object like this:

```js
{
	id: {
		enabled: true,
		toggle: function () {}
	},
	anotherId: {
		enabled: false,
		toggle: function () {}
	}
}
```

And that object is provided by this: #1054, which you can merge here.

Reply:
@cwaring have got an idea. We can do the following, when `desktop: true`:

```js
// Check if we should show it
store.doDesktopShouldShowExperiment(experimentId) // ==> true or false

// Get experiment value
store.selectDesktopSettings().experiments.experimentId // ==> true or false

// Toggle experiment value
store.doDesktopToggleExperiment(experimentId) // async action, do not wait for it
```

Sounds good? Take that in account and make all the changes you need, I'll make those functions work on #1054 and then merge it here, sounds good?

Reply:
I think we can do something to connect the two here, my concern is that I wouldn't want  to blindly toggle and then update the enabled state in case there is an error so perhaps we need a result method in the reducer here and the toggle action simply dispatches call to toggle?

Reply:
Also `enabled` in this context means that the experiment is activated not available to be activated, it's persisted in the state

Reply:
@cwaring it seems you read the comment I marked as outdated a while ago. It's not you who needs to update the `enabled` value. IPFS Desktop takes care of that, not Web UI. Please see the latest comment (https://github.com/ipfs-shipyard/ipfs-webui/pull/1048#discussion_r290652415). Perhaps it's a better idea

Reply:
I'm not sure what happened there. I didn't mark it as outdated, possibly because we replied at the same time? Good ideas but I'm still not totally clear, I'll look into this in more detail when I'm back a the desk soon 👌

Reply:
@cwaring cool! I opened a PR: https://github.com/ipfs-shipyard/ipfs-webui/pull/1055. Feel free to merge it here. It is all explained there.